From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Glitch <glitch.g3431@gmail.com>
Date: Sat, 12 Mar 2022 21:08:03 -0600
Subject: [PATCH] Rebrand to Quilt and set up publishing for our maven.

This includes:
- Changing the plugin marker, including in the tests
- Using a different set of caches
- Using the Quilt Loader system properties
- Use quilt loader in all tests
- Using `quilt_installer.json`

diff --git a/bootstrap/test-project/build.gradle b/bootstrap/test-project/build.gradle
index 561915428a80131439ea9e1d9f12804c491d399a..db648181f772520d1607ce2510a66be1d00167c2 100644
--- a/bootstrap/test-project/build.gradle
+++ b/bootstrap/test-project/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom' version '0.13.local'
+	id 'org.quiltmc.loom' version '0.13.local'
 }
 
 dependencies {
diff --git a/build.gradle b/build.gradle
index 10a6e0c0dc68d863684b37e904396fb7b1cab39f..1c80c3c8b75a347daab1b6f6f5e81de778c47e8f 100644
--- a/build.gradle
+++ b/build.gradle
@@ -23,7 +23,7 @@ tasks.withType(org.jetbrains.kotlin.gradle.tasks.KotlinCompile).all {
 	}
 }
 
-group = 'net.fabricmc'
+group = 'org.quiltmc'
 archivesBaseName = project.name
 def baseVersion = '1.1'
 
@@ -170,8 +170,8 @@ codenarc {
 
 gradlePlugin {
 	plugins {
-		fabricLoom {
-			id = 'fabric-loom'
+		quiltLoom {
+			id = 'org.quiltmc.loom'
 			implementationClass = 'net.fabricmc.loom.bootstrap.LoomGradlePluginBootstrap'
 		}
 	}
@@ -227,8 +227,8 @@ publishing {
 
 		// Manually crate the plugin marker for snapshot versions
 		snapshotPlugin(MavenPublication) { publication ->
-			groupId 'fabric-loom'
-			artifactId 'fabric-loom.gradle.plugin'
+			groupId 'org.quiltmc.loom'
+			artifactId 'org.quiltmc.loom.gradle.plugin'
 			version baseVersion + '-SNAPSHOT'
 
 			pom.withXml({
@@ -238,9 +238,9 @@ publishing {
 				Node dependencies = root.appendChild(document.createElement('dependencies'))
 				Node dependency = dependencies.appendChild(document.createElement('dependency'))
 				Node groupId = dependency.appendChild(document.createElement('groupId'))
-				groupId.setTextContent('net.fabricmc')
+				groupId.setTextContent('org.quiltmc')
 				Node artifactId = dependency.appendChild(document.createElement('artifactId'))
-				artifactId.setTextContent('fabric-loom')
+				artifactId.setTextContent('loom')
 				Node version = dependency.appendChild(document.createElement('version'))
 				version.setTextContent(baseVersion + '-SNAPSHOT')
 			})
@@ -248,6 +248,7 @@ publishing {
 	}
 	repositories {
 		maven {
+			name = 'release'
 			if (ENV.MAVEN_URL) {
 				url ENV.MAVEN_URL
 				credentials {
@@ -256,6 +257,27 @@ publishing {
 				}
 			}
 		}
+		maven {
+			name = "snapshot"
+			if (ENV.SNAPSHOTS_URL) {
+				url ENV.SNAPSHOTS_URL
+				credentials {
+					username ENV.SNAPSHOTS_USERNAME
+					password ENV.SNAPSHOTS_PASSWORD
+				}
+			}
+		}
+	}
+}
+
+tasks.withType(PublishToMavenRepository) {
+	onlyIf {
+		(repository == publishing.repositories.release && publication == publishing.publications.plugin)
+				||
+				(repository == publishing.repositories.snapshot
+						&& (publication == publishing.publications.snapshot
+						|| publication == publishing.publications.snapshotPlugin))
+				|| (repository == publishing.repositories.release && publication.name == "quiltLoomPluginMarkerMaven")
 	}
 }
 
diff --git a/gradle.properties b/gradle.properties
index 31014b156127886c44921900f480e68d4528da25..ee1d4c3083cdb729df09522e88a41351d332a875 100644
--- a/gradle.properties
+++ b/gradle.properties
@@ -1,5 +1,5 @@
-name = fabric-loom
-description = The Gradle plugin for Fabric
-url = https://github.com/FabricMC/fabric-loom
+name = loom
+description = The Gradle plugin for Quilt
+url = https://github.com/QuiltMC/quilt-loom
 
 kotlin.stdlib.default.dependency = false
\ No newline at end of file
diff --git a/src/main/java/net/fabricmc/loom/LoomGradlePlugin.java b/src/main/java/net/fabricmc/loom/LoomGradlePlugin.java
index a75b67fcbd76a74b9e6f4786dd5bc5e19690790b..49fc762edec712bec2f51cadd0dc95d0f7787c90 100644
--- a/src/main/java/net/fabricmc/loom/LoomGradlePlugin.java
+++ b/src/main/java/net/fabricmc/loom/LoomGradlePlugin.java
@@ -62,7 +62,7 @@ public class LoomGradlePlugin implements BootstrappedPlugin {
 	}
 
 	public void apply(Project project) {
-		project.getLogger().lifecycle("Fabric Loom: " + LOOM_VERSION);
+		project.getLogger().lifecycle("Quilt Loom: " + LOOM_VERSION);
 		LibraryLocationLogger.logLibraryVersions();
 
 		// Apply default plugins
diff --git a/src/main/java/net/fabricmc/loom/configuration/LoomDependencyManager.java b/src/main/java/net/fabricmc/loom/configuration/LoomDependencyManager.java
index b41b39bf86c6bfcd8c600eb70d160d801e66f510..7d52086caedb771641e5355abddb3cb8f2345036 100644
--- a/src/main/java/net/fabricmc/loom/configuration/LoomDependencyManager.java
+++ b/src/main/java/net/fabricmc/loom/configuration/LoomDependencyManager.java
@@ -44,7 +44,7 @@ public class LoomDependencyManager {
 		sourceRemapper.remapAll();
 
 		if (extension.getInstallerData() == null) {
-			project.getLogger().warn("fabric-installer.json not found in dependencies!");
+			project.getLogger().warn("quilt_installer.json not found in dependencies!");
 		}
 	}
 }
diff --git a/src/main/java/net/fabricmc/loom/extension/LoomFilesBaseImpl.java b/src/main/java/net/fabricmc/loom/extension/LoomFilesBaseImpl.java
index d7f0c469b92b3acfb24a623306f8d63bf44eafbe..12f19648c39d6eec91d8bd09cdd3400aadac3ca3 100644
--- a/src/main/java/net/fabricmc/loom/extension/LoomFilesBaseImpl.java
+++ b/src/main/java/net/fabricmc/loom/extension/LoomFilesBaseImpl.java
@@ -50,22 +50,22 @@ public abstract class LoomFilesBaseImpl implements LoomFiles {
 
 	@Override
 	public File getUserCache() {
-		return createFile(getGradleUserHomeDir(), "caches" + File.separator + "fabric-loom");
+		return createFile(getGradleUserHomeDir(), "caches" + File.separator + "quilt-loom");
 	}
 
 	@Override
 	public File getRootProjectPersistentCache() {
-		return createFile(getRootDir(), ".gradle" + File.separator + "loom-cache");
+		return createFile(getRootDir(), ".gradle" + File.separator + "quilt-loom-cache");
 	}
 
 	@Override
 	public File getProjectPersistentCache() {
-		return createFile(getProjectDir(), ".gradle" + File.separator + "loom-cache");
+		return createFile(getProjectDir(), ".gradle" + File.separator + "quilt-loom-cache");
 	}
 
 	@Override
 	public File getProjectBuildCache() {
-		return createFile(getBuildDir(), "loom-cache");
+		return createFile(getBuildDir(), "quilt-loom-cache");
 	}
 
 	@Override
diff --git a/src/main/java/net/fabricmc/loom/task/launch/GenerateDLIConfigTask.java b/src/main/java/net/fabricmc/loom/task/launch/GenerateDLIConfigTask.java
index 7176e985cb697a6f8e4bbb5372908b03642bf852..0439eea2bd5d80aa236f9e232dd5f21310548841 100644
--- a/src/main/java/net/fabricmc/loom/task/launch/GenerateDLIConfigTask.java
+++ b/src/main/java/net/fabricmc/loom/task/launch/GenerateDLIConfigTask.java
@@ -55,15 +55,17 @@ public abstract class GenerateDLIConfigTask extends AbstractLoomTask {
 		}
 
 		final LaunchConfig launchConfig = new LaunchConfig()
-				.property("fabric.development", "true")
-				.property("fabric.remapClasspathFile", getExtension().getFiles().getRemapClasspathFile().getAbsolutePath())
+				.property("loader.development", "true")
+				.property("loader.remapClasspathFile", getExtension().getFiles().getRemapClasspathFile().getAbsolutePath())
 				.property("log4j.configurationFile", getAllLog4JConfigFiles())
 				.property("log4j2.formatMsgNoLookups", "true")
 
 				.argument("client", "--assetIndex")
 				.argument("client", getExtension().getMinecraftProvider().getVersionInfo().assetIndex().fabricId(getExtension().getMinecraftProvider().minecraftVersion()))
 				.argument("client", "--assetsDir")
-				.argument("client", assetsDirectory.getAbsolutePath());
+				.argument("client", assetsDirectory.getAbsolutePath())
+				.argument("client", "--version")
+				.argument("client", "Quilt Loom");
 
 		if (versionInfo.hasNativesToExtract()) {
 			String nativesPath = getExtension().getFiles().getNativesDirectory(getProject()).getAbsolutePath();
@@ -74,12 +76,12 @@ public abstract class GenerateDLIConfigTask extends AbstractLoomTask {
 		}
 
 		if (getExtension().areEnvironmentSourceSetsSplit()) {
-			launchConfig.property("client", "fabric.gameJarPath.client", getGameJarPath("client"));
-			launchConfig.property("fabric.gameJarPath", getGameJarPath("common"));
+			launchConfig.property("client", "loader.gameJarPath.client", getGameJarPath("client"));
+			launchConfig.property("loader.gameJarPath", getGameJarPath("common"));
 		}
 
 		if (!getExtension().getMods().isEmpty()) {
-			launchConfig.property("fabric.classPathGroups", getClassPathGroups());
+			launchConfig.property("loader.classPathGroups", getClassPathGroups());
 		}
 
 		final boolean plainConsole = getProject().getGradle().getStartParameter().getConsoleOutput() == ConsoleOutput.Plain;
diff --git a/src/main/java/net/fabricmc/loom/task/service/SourceRemapperService.java b/src/main/java/net/fabricmc/loom/task/service/SourceRemapperService.java
index b3f5456c1317acc23b903df9451b59683d677dcf..18341c50640d4e42f2391ae46c3d2be772c3a18f 100644
--- a/src/main/java/net/fabricmc/loom/task/service/SourceRemapperService.java
+++ b/src/main/java/net/fabricmc/loom/task/service/SourceRemapperService.java
@@ -88,7 +88,7 @@ public final class SourceRemapperService implements SharedService {
 		// Create a temp directory with all of the sources
 		if (!Files.isDirectory(source)) {
 			isSrcTmp = true;
-			srcPath = Files.createTempDirectory("fabric-loom-src");
+			srcPath = Files.createTempDirectory("quilt-loom-src");
 			ZipUtils.unpackAll(source, srcPath);
 		}
 
diff --git a/src/main/java/net/fabricmc/loom/util/SourceRemapper.java b/src/main/java/net/fabricmc/loom/util/SourceRemapper.java
index e3348689d7eb112338862c164be8bbbfa8e22d03..2a206d5396b8e304ad8d0d65b2b73700d4373f8f 100644
--- a/src/main/java/net/fabricmc/loom/util/SourceRemapper.java
+++ b/src/main/java/net/fabricmc/loom/util/SourceRemapper.java
@@ -128,7 +128,7 @@ public class SourceRemapper {
 		if (!source.isDirectory()) {
 			// create tmp directory
 			isSrcTmp = true;
-			srcPath = Files.createTempDirectory("fabric-loom-src");
+			srcPath = Files.createTempDirectory("quilt-loom-src");
 			ZipUtils.unpackAll(source.toPath(), srcPath);
 		}
 
diff --git a/src/test/groovy/net/fabricmc/loom/test/integration/CustomManifestTest.groovy b/src/test/groovy/net/fabricmc/loom/test/integration/CustomManifestTest.groovy
index eb732700eb1ab74706e476aca2e79a535194edb1..814a0e19d0b95b7e292444f8d2babc7d40bc71bf 100644
--- a/src/test/groovy/net/fabricmc/loom/test/integration/CustomManifestTest.groovy
+++ b/src/test/groovy/net/fabricmc/loom/test/integration/CustomManifestTest.groovy
@@ -44,7 +44,7 @@ class CustomManifestTest extends Specification implements GradleProjectTestTrait
                 dependencies {
                     minecraft "com.mojang:minecraft:1.18_experimental-snapshot-1"
                     mappings "net.fabricmc:yarn:1.18_experimental-snapshot-1+build.2:v2"
-                    modImplementation "net.fabricmc:fabric-loader:0.11.6"
+                    modImplementation "org.quiltmc:quilt-loader:0.16.0-beta.5"
                 }
             '''
         when:
diff --git a/src/test/groovy/net/fabricmc/loom/test/integration/ExperimentalVersionsTest.groovy b/src/test/groovy/net/fabricmc/loom/test/integration/ExperimentalVersionsTest.groovy
index a95073e5e6294f0197f21ffb2c6b1047b4051ef2..ade0cc3282b965fb740aaade1ccc45815efc1f64 100644
--- a/src/test/groovy/net/fabricmc/loom/test/integration/ExperimentalVersionsTest.groovy
+++ b/src/test/groovy/net/fabricmc/loom/test/integration/ExperimentalVersionsTest.groovy
@@ -41,7 +41,7 @@ class ExperimentalVersionsTest extends Specification implements GradleProjectTes
                 dependencies {
                     minecraft "com.mojang:minecraft:1.18_experimental-snapshot-1"
                     mappings "net.fabricmc:yarn:1.18_experimental-snapshot-1+build.2:v2"
-                    modImplementation "net.fabricmc:fabric-loader:0.11.6"
+                    modImplementation "org.quiltmc:quilt-loader:0.16.0-beta.5"
                 }
             '''
 
diff --git a/src/test/groovy/net/fabricmc/loom/test/integration/LegacyProjectTest.groovy b/src/test/groovy/net/fabricmc/loom/test/integration/LegacyProjectTest.groovy
index 7461566c29a5e7d63ff2f48b7c45c0ea9db0a665..9d12f23af7c78b5dcf7b9956b5bec6f74e9284a7 100644
--- a/src/test/groovy/net/fabricmc/loom/test/integration/LegacyProjectTest.groovy
+++ b/src/test/groovy/net/fabricmc/loom/test/integration/LegacyProjectTest.groovy
@@ -63,7 +63,7 @@ class LegacyProjectTest extends Specification implements GradleProjectTestTrait
 						// No names
 					}
 
-                    modImplementation "net.fabricmc:fabric-loader:0.12.12"
+                    modImplementation "org.quiltmc:quilt-loader:0.16.0-beta.5"
                 }
 			"""
 
@@ -101,7 +101,7 @@ class LegacyProjectTest extends Specification implements GradleProjectTestTrait
 						// No names
 					}
 
-                    modImplementation "net.fabricmc:fabric-loader:0.12.12"
+                    modImplementation "org.quiltmc:quilt-loader:0.16.0-beta.5"
                 }
 			"""
 
diff --git a/src/test/groovy/net/fabricmc/loom/test/integration/MCJarConfigTest.groovy b/src/test/groovy/net/fabricmc/loom/test/integration/MCJarConfigTest.groovy
index 66deb25fc0fcdd0ae5e7e2a6f405ebd0fbe957e7..1a3bc61de50aa5782fbdbbee8cea308eecfe568a 100644
--- a/src/test/groovy/net/fabricmc/loom/test/integration/MCJarConfigTest.groovy
+++ b/src/test/groovy/net/fabricmc/loom/test/integration/MCJarConfigTest.groovy
@@ -45,7 +45,7 @@ class MCJarConfigTest extends Specification implements GradleProjectTestTrait {
                 dependencies {
                     minecraft "com.mojang:minecraft:1.18.1"
                     mappings "net.fabricmc:yarn:1.18.1+build.18:v2"
-                    modImplementation "net.fabricmc:fabric-loader:0.12.12"
+                    modImplementation "org.quiltmc:quilt-loader:0.16.0-beta.5"
                 }
             '''
 
@@ -72,7 +72,7 @@ class MCJarConfigTest extends Specification implements GradleProjectTestTrait {
                 dependencies {
                     minecraft "com.mojang:minecraft:1.18.1"
                     mappings "net.fabricmc:yarn:1.18.1+build.18:v2"
-                    modImplementation "net.fabricmc:fabric-loader:0.12.12"
+                    modImplementation "org.quiltmc:quilt-loader:0.16.0-beta.5"
                 }
             '''
 
@@ -99,7 +99,7 @@ class MCJarConfigTest extends Specification implements GradleProjectTestTrait {
                 dependencies {
                     minecraft "com.mojang:minecraft:1.18.1"
                     mappings "net.fabricmc:yarn:1.18.1+build.18:v2"
-                    modImplementation "net.fabricmc:fabric-loader:0.12.12"
+                    modImplementation "org.quiltmc:quilt-loader:0.16.0-beta.5"
                 }
             '''
 
@@ -126,7 +126,7 @@ class MCJarConfigTest extends Specification implements GradleProjectTestTrait {
                 dependencies {
                     minecraft "com.mojang:minecraft:1.18.1"
                     mappings "net.fabricmc:yarn:1.18.1+build.18:v2"
-                    modImplementation "net.fabricmc:fabric-loader:0.12.12"
+                    modImplementation "org.quiltmc:quilt-loader:0.16.0-beta.5"
                 }
             '''
 
diff --git a/src/test/groovy/net/fabricmc/loom/test/integration/RunConfigTest.groovy b/src/test/groovy/net/fabricmc/loom/test/integration/RunConfigTest.groovy
index 18f8e684ec06195d143c290e07c3c0c8906e1e3f..152baa8b73b06dfac3fa18943004109321944e28 100644
--- a/src/test/groovy/net/fabricmc/loom/test/integration/RunConfigTest.groovy
+++ b/src/test/groovy/net/fabricmc/loom/test/integration/RunConfigTest.groovy
@@ -68,7 +68,7 @@ class RunConfigTest extends Specification implements GradleProjectTestTrait {
                 dependencies {
                     minecraft "com.mojang:minecraft:1.18.1"
                     mappings "net.fabricmc:yarn:1.18.1+build.18:v2"
-                    modImplementation "net.fabricmc:fabric-loader:0.12.12"
+                    modImplementation "org.quiltmc:quilt-loader:0.16.0-beta.5"
                 }
             '''
 
diff --git a/src/test/groovy/net/fabricmc/loom/test/integration/SimpleProjectTest.groovy b/src/test/groovy/net/fabricmc/loom/test/integration/SimpleProjectTest.groovy
index ced58f6bbf2ec686dd59432b0972936d8e5e9e02..f0f35b03c13eb6ac7683e82d97f619f91f7791c5 100644
--- a/src/test/groovy/net/fabricmc/loom/test/integration/SimpleProjectTest.groovy
+++ b/src/test/groovy/net/fabricmc/loom/test/integration/SimpleProjectTest.groovy
@@ -50,7 +50,7 @@ class SimpleProjectTest extends Specification implements GradleProjectTestTrait
 			def serverResult = server.run()
 		then:
 			result.task(":build").outcome == SUCCESS
-			gradle.getOutputZipEntry("fabric-example-mod-1.0.0.jar", "META-INF/MANIFEST.MF").contains("Fabric-Loom-Version: 0.0.0+unknown")
+			gradle.getOutputZipEntry("fabric-example-mod-1.0.0.jar", "META-INF/MANIFEST.MF").contains("Fabric-Loom-Version: 0.12.0+unknown")
 			gradle.getOutputZipEntry("fabric-example-mod-1.0.0-sources.jar", "net/fabricmc/example/mixin/ExampleMixin.java").contains("class_442") // Very basic test to ensure sources got remapped
 
 			serverResult.successful()
diff --git a/src/test/groovy/net/fabricmc/loom/test/util/GradleProjectTestTrait.groovy b/src/test/groovy/net/fabricmc/loom/test/util/GradleProjectTestTrait.groovy
index ad590af4e02f480fa5b84d6cdd09ec4cbc454d5e..5767b9402ba1873a4dea48e8413fd2eb2e0248c5 100644
--- a/src/test/groovy/net/fabricmc/loom/test/util/GradleProjectTestTrait.groovy
+++ b/src/test/groovy/net/fabricmc/loom/test/util/GradleProjectTestTrait.groovy
@@ -240,7 +240,7 @@ trait GradleProjectTestTrait {
         }
 
         File getGeneratedLocalSources(String mappings) {
-            return new File(getProjectDir(), ".gradle/loom-cache/minecraftMaven/net/minecraft/minecraft-merged-project-root/${mappings}/minecraft-merged-project-root-${mappings}-sources.jar")
+            return new File(getProjectDir(), ".gradle/quilt-loom-cache/minecraftMaven/net/minecraft/minecraft-merged-project-root/${mappings}/minecraft-merged-project-root-${mappings}-sources.jar")
         }
 
         void buildSrc(String name) {
diff --git a/src/test/resources/projects/accesswidener/build.gradle b/src/test/resources/projects/accesswidener/build.gradle
index 9fa3cc36ef23ee7f9a03120624596c6fe1788de4..c3950e8fbcc865a12a2a5a5ededf0fbddf3e6698 100644
--- a/src/test/resources/projects/accesswidener/build.gradle
+++ b/src/test/resources/projects/accesswidener/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'org.quiltmc.loom'
 	id 'maven-publish'
 }
 
@@ -26,7 +26,7 @@ dependencies {
 	// To change the versions see the gradle.properties file
 	minecraft "com.mojang:minecraft:${project.minecraft_version}"
 	mappings "net.fabricmc:yarn:${project.yarn_mappings}:v2"
-	modImplementation "net.fabricmc:fabric-loader:${project.loader_version}"
+	modImplementation "org.quiltmc:quilt-loader:${project.loader_version}"
 
 	// Fabric API. This is technically optional, but you probably want it anyway.
 	modImplementation "net.fabricmc.fabric-api:fabric-api:${project.fabric_version}"
diff --git a/src/test/resources/projects/accesswidener/gradle.properties b/src/test/resources/projects/accesswidener/gradle.properties
index 2e85dfd8d7d617d3e8d5c71e0eeab4c93891edbf..970667432831ef16d900cbb1822786e61c804260 100644
--- a/src/test/resources/projects/accesswidener/gradle.properties
+++ b/src/test/resources/projects/accesswidener/gradle.properties
@@ -5,7 +5,7 @@ org.gradle.jvmargs=-Xmx1G
 	# check these on https://fabricmc.net/use
 	minecraft_version=1.16.5
 	yarn_mappings=1.16.5+build.5
-	loader_version=0.11.2
+	loader_version=0.16.0-beta.16
 
 # Mod Properties
 	mod_version = 1.0.0
diff --git a/src/test/resources/projects/decompile/build.gradle b/src/test/resources/projects/decompile/build.gradle
index 993ac667071e4e3e2b1a8301121e817950663d61..17961497657cb80f2ce1cb2463f1a28c11e85c5d 100644
--- a/src/test/resources/projects/decompile/build.gradle
+++ b/src/test/resources/projects/decompile/build.gradle
@@ -1,11 +1,11 @@
 plugins {
-	id 'fabric-loom'
+	id 'org.quiltmc.loom'
 }
 
 dependencies {
 	minecraft "com.mojang:minecraft:21w38a"
 	mappings "net.fabricmc:yarn:21w38a+build.11:v2"
-	modImplementation "net.fabricmc:fabric-loader:0.11.7"
+	modImplementation "org.quiltmc:quilt-loader:0.16.0-beta.5"
 }
 
 loom {
diff --git a/src/test/resources/projects/dependencyResolutionManagement/basic/build.gradle b/src/test/resources/projects/dependencyResolutionManagement/basic/build.gradle
index c7b6ef6c0a20c1d4b2936dcbdf222f201be41ccc..d0df797cc20ddee83098186a731366663fc2d191 100644
--- a/src/test/resources/projects/dependencyResolutionManagement/basic/build.gradle
+++ b/src/test/resources/projects/dependencyResolutionManagement/basic/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'org.quiltmc.loom'
 	id 'maven-publish'
 }
 
@@ -22,7 +22,7 @@ dependencies {
 	// To change the versions see the gradle.properties file
 	minecraft "com.mojang:minecraft:${project.minecraft_version}"
 	mappings "net.fabricmc:yarn:${project.yarn_mappings}:v2"
-	modImplementation "net.fabricmc:fabric-loader:${project.loader_version}"
+	modImplementation "org.quiltmc:quilt-loader:${project.loader_version}"
 
 	// Fabric API. This is technically optional, but you probably want it anyway.
 	modImplementation "net.fabricmc.fabric-api:fabric-api:${project.fabric_version}"
diff --git a/src/test/resources/projects/dependencyResolutionManagement/basic/gradle.properties b/src/test/resources/projects/dependencyResolutionManagement/basic/gradle.properties
index 40e9e5717128a126a91dcfcc0b6fd62fb9100792..523f9a2e212924f27dd2c6f66305b761c3e0c247 100644
--- a/src/test/resources/projects/dependencyResolutionManagement/basic/gradle.properties
+++ b/src/test/resources/projects/dependencyResolutionManagement/basic/gradle.properties
@@ -5,7 +5,7 @@ org.gradle.jvmargs=-Xmx1G
 	# check these on https://fabricmc.net/use
 	minecraft_version=1.16.5
 	yarn_mappings=1.16.5+build.6
-	loader_version=0.11.3
+	loader_version=0.16.0-beta.16
 
 # Mod Properties
 	mod_version = 1.0.0
diff --git a/src/test/resources/projects/dependencyResolutionManagement/projmap/build.gradle b/src/test/resources/projects/dependencyResolutionManagement/projmap/build.gradle
index 6472ec7e63ab88f17b626c68393faca58042a8d2..6c8ee6bd10d606f86e075e2b094d5414c0075577 100644
--- a/src/test/resources/projects/dependencyResolutionManagement/projmap/build.gradle
+++ b/src/test/resources/projects/dependencyResolutionManagement/projmap/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'org.quiltmc.loom'
 	id 'maven-publish'
 }
 
@@ -22,7 +22,7 @@ dependencies {
 	// To change the versions see the gradle.properties file
 	minecraft "com.mojang:minecraft:${project.minecraft_version}"
 	mappings "net.fabricmc:yarn:${project.yarn_mappings}:v2"
-	modImplementation "net.fabricmc:fabric-loader:${project.loader_version}"
+	modImplementation "org.quiltmc:quilt-loader:${project.loader_version}"
 
 	// Fabric API. This is technically optional, but you probably want it anyway.
 	modImplementation "net.fabricmc.fabric-api:fabric-api:${project.fabric_version}"
diff --git a/src/test/resources/projects/dependencyResolutionManagement/projmap/gradle.properties b/src/test/resources/projects/dependencyResolutionManagement/projmap/gradle.properties
index 40e9e5717128a126a91dcfcc0b6fd62fb9100792..523f9a2e212924f27dd2c6f66305b761c3e0c247 100644
--- a/src/test/resources/projects/dependencyResolutionManagement/projmap/gradle.properties
+++ b/src/test/resources/projects/dependencyResolutionManagement/projmap/gradle.properties
@@ -5,7 +5,7 @@ org.gradle.jvmargs=-Xmx1G
 	# check these on https://fabricmc.net/use
 	minecraft_version=1.16.5
 	yarn_mappings=1.16.5+build.6
-	loader_version=0.11.3
+	loader_version=0.16.0-beta.16
 
 # Mod Properties
 	mod_version = 1.0.0
diff --git a/src/test/resources/projects/dependencyResolutionManagement/settings.gradle b/src/test/resources/projects/dependencyResolutionManagement/settings.gradle
index c05e369995a4d8c5479de657d6c8ab52a6cba940..6b77297f004043508453a0907ea19bca3a372c5e 100644
--- a/src/test/resources/projects/dependencyResolutionManagement/settings.gradle
+++ b/src/test/resources/projects/dependencyResolutionManagement/settings.gradle
@@ -1,11 +1,11 @@
 pluginManagement {
     plugins {
-        id 'fabric-loom'
+        id 'org.quiltmc.loom'
     }
 }
 
 plugins {
-    id 'fabric-loom'
+    id 'org.quiltmc.loom'
 }
 
 dependencyResolutionManagement {
diff --git a/src/test/resources/projects/interfaceInjection/build.gradle b/src/test/resources/projects/interfaceInjection/build.gradle
index 2396f3c34efc5d82e7af392cfa52ef420e5735fd..83e677a053b55621c93b843c012f23e84b79e7d2 100644
--- a/src/test/resources/projects/interfaceInjection/build.gradle
+++ b/src/test/resources/projects/interfaceInjection/build.gradle
@@ -1,7 +1,7 @@
 // This is used by a range of tests that append to this file before running the gradle tasks.
 // Can be used for tests that require minimal custom setup
 plugins {
-	id 'fabric-loom'
+	id 'org.quiltmc.loom'
 	id 'maven-publish'
 }
 
@@ -12,7 +12,7 @@ group = "com.example"
 dependencies {
 	minecraft "com.mojang:minecraft:1.17.1"
 	mappings "net.fabricmc:yarn:1.17.1+build.59:v2"
-	modImplementation "net.fabricmc:fabric-loader:0.11.6"
+	modImplementation "org.quiltmc:quilt-loader:0.16.0-beta.5"
 
 	// I-faces will still be applied as the jar is on both configurations.
 	modCompileOnly files("dummy.jar")
diff --git a/src/test/resources/projects/java16/build.gradle b/src/test/resources/projects/java16/build.gradle
index 45acf714efb54893e7867807d3bed47aa1a58223..e2bfaeebb5df8090259eb9d18ebdf28eddf66504 100644
--- a/src/test/resources/projects/java16/build.gradle
+++ b/src/test/resources/projects/java16/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'org.quiltmc.loom'
 	id 'maven-publish'
 }
 
@@ -18,7 +18,7 @@ dependencies {
 	// To change the versions see the gradle.properties file
 	minecraft "com.mojang:minecraft:${project.minecraft_version}"
 	mappings "net.fabricmc:yarn:${project.yarn_mappings}:v2"
-	modImplementation "net.fabricmc:fabric-loader:${project.loader_version}"
+	modImplementation "org.quiltmc:quilt-loader:${project.loader_version}"
 	modImplementation "net.fabricmc.fabric-api:fabric-api:${project.fabric_version}"
 }
 
diff --git a/src/test/resources/projects/java16/gradle.properties b/src/test/resources/projects/java16/gradle.properties
index cdc647ce754f229ba0daa586be4031a0b734251e..681c2d0af7e67f717c0f28bda66a911da8b2490e 100644
--- a/src/test/resources/projects/java16/gradle.properties
+++ b/src/test/resources/projects/java16/gradle.properties
@@ -2,7 +2,7 @@ org.gradle.jvmargs=-Xmx1G
 
 minecraft_version=1.18-pre1
 yarn_mappings=1.18-pre1+build.2
-loader_version=0.12.5
+loader_version=0.16.0-beta.16
 fabric_version=0.42.1+1.18
 
 mod_version = 1.0.0
diff --git a/src/test/resources/projects/kotlin/build.gradle.kts b/src/test/resources/projects/kotlin/build.gradle.kts
index 7f2e0f5d28bd402223319574cf804cc7ddeabc62..8175dce88d1b8f226dbb4b2db58744cfd458870c 100644
--- a/src/test/resources/projects/kotlin/build.gradle.kts
+++ b/src/test/resources/projects/kotlin/build.gradle.kts
@@ -3,7 +3,7 @@ import java.util.Properties
 plugins {
 	kotlin("jvm") version "1.7.22"
 	kotlin("plugin.serialization") version "1.7.22"
-	id("fabric-loom")
+	id("org.quiltmc.loom")
 	`maven-publish`
 }
 
@@ -18,7 +18,7 @@ version = "0.0.1"
 dependencies {
 	minecraft(group = "com.mojang", name = "minecraft", version = "1.16.5")
 	mappings(group = "net.fabricmc", name = "yarn", version = "1.16.5+build.5", classifier = "v2")
-	modImplementation("net.fabricmc:fabric-loader:0.12.12")
+	modImplementation("org.quiltmc:quilt-loader:0.16.1")
 	modImplementation(group = "net.fabricmc", name = "fabric-language-kotlin", version = "1.8.7+kotlin.1.7.22")
 }
 
diff --git a/src/test/resources/projects/legacy/build.gradle b/src/test/resources/projects/legacy/build.gradle
index 3b460c49aaa95a8b98cff43df820b257498ad668..2cd1bdbf885b93a39bccf1300597a0c57d8c1d45 100644
--- a/src/test/resources/projects/legacy/build.gradle
+++ b/src/test/resources/projects/legacy/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'org.quiltmc.loom'
 }
 
 sourceCompatibility = JavaVersion.VERSION_1_8
@@ -12,7 +12,7 @@ group = project.maven_group
 dependencies {
 	minecraft "com.mojang:minecraft:${project.minecraft_version}"
 	mappings "net.fabricmc:yarn:${project.yarn_mappings}"
-	modImplementation "net.fabricmc:fabric-loader:${project.loader_version}"
+	modImplementation "org.quiltmc:quilt-loader:${project.loader_version}"
 	modImplementation "net.fabricmc.fabric-api:fabric-api:${project.fabric_version}"
 }
 
diff --git a/src/test/resources/projects/legacy/gradle.properties b/src/test/resources/projects/legacy/gradle.properties
index 440dabe5b015628affe10c53ea2a8fcb977e9409..aba0090df90da5eae231db6dc1c61040a7d3d029 100644
--- a/src/test/resources/projects/legacy/gradle.properties
+++ b/src/test/resources/projects/legacy/gradle.properties
@@ -5,7 +5,7 @@ org.gradle.jvmargs=-Xmx1G
 	# check these on https://fabricmc.net/use
 	minecraft_version=1.14.4
 	yarn_mappings=1.14.4+build.18
-	loader_version=0.11.2
+	loader_version=0.16.0-beta.16
 
 # Mod Properties
 	mod_version = 1.0.0
diff --git a/src/test/resources/projects/localFileDependency/build.gradle b/src/test/resources/projects/localFileDependency/build.gradle
index 811d945a9a903d2e43e949de2e2fea4f99882edd..36a99bc5fa0dc9854fe8bf740bf989550a8f7510 100644
--- a/src/test/resources/projects/localFileDependency/build.gradle
+++ b/src/test/resources/projects/localFileDependency/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'org.quiltmc.loom'
 	id 'maven-publish'
 }
 
@@ -25,7 +25,7 @@ dependencies {
 	// To change the versions see the gradle.properties file
 	minecraft "com.mojang:minecraft:${project.minecraft_version}"
 	mappings "net.fabricmc:yarn:${project.yarn_mappings}:v2"
-	modImplementation "net.fabricmc:fabric-loader:${project.loader_version}"
+	modImplementation "org.quiltmc:quilt-loader:${project.loader_version}"
 
 	// Fabric API. This is technically optional, but you probably want it anyway.
 	modImplementation "net.fabricmc.fabric-api:fabric-api:${project.fabric_version}"
diff --git a/src/test/resources/projects/localFileDependency/gradle.properties b/src/test/resources/projects/localFileDependency/gradle.properties
index 2e85dfd8d7d617d3e8d5c71e0eeab4c93891edbf..970667432831ef16d900cbb1822786e61c804260 100644
--- a/src/test/resources/projects/localFileDependency/gradle.properties
+++ b/src/test/resources/projects/localFileDependency/gradle.properties
@@ -5,7 +5,7 @@ org.gradle.jvmargs=-Xmx1G
 	# check these on https://fabricmc.net/use
 	minecraft_version=1.16.5
 	yarn_mappings=1.16.5+build.5
-	loader_version=0.11.2
+	loader_version=0.16.0-beta.16
 
 # Mod Properties
 	mod_version = 1.0.0
diff --git a/src/test/resources/projects/localRuntime/build.gradle b/src/test/resources/projects/localRuntime/build.gradle
index 18f4f0db4b857098a782cfa548a2954c44f363dc..c119c0085e59b282c1fc10335aa1dbb0d7cfb2e3 100644
--- a/src/test/resources/projects/localRuntime/build.gradle
+++ b/src/test/resources/projects/localRuntime/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'org.quiltmc.loom'
 	id 'maven-publish'
 }
 
@@ -22,7 +22,7 @@ dependencies {
 	// To change the versions see the gradle.properties file
 	minecraft "com.mojang:minecraft:${project.minecraft_version}"
 	mappings "net.fabricmc:yarn:${project.yarn_mappings}:v2"
-	modImplementation "net.fabricmc:fabric-loader:${project.loader_version}"
+	modImplementation "org.quiltmc:quilt-loader:${project.loader_version}"
 
 	// Fabric API. This is technically optional, but you probably want it anyway.
 	modLocalRuntime "net.fabricmc.fabric-api:fabric-api:${project.fabric_version}"
diff --git a/src/test/resources/projects/localRuntime/gradle.properties b/src/test/resources/projects/localRuntime/gradle.properties
index 845c356072dbb73ddb31b97826e8cb4fbaf138e0..c01c88366c6038008ee1d2f668b63bc2de2e138a 100644
--- a/src/test/resources/projects/localRuntime/gradle.properties
+++ b/src/test/resources/projects/localRuntime/gradle.properties
@@ -5,7 +5,7 @@ org.gradle.jvmargs=-Xmx1G
 	# check these on https://fabricmc.net/use
 	minecraft_version=1.16.5
 	yarn_mappings=1.16.5+build.5
-	loader_version=0.11.2
+	loader_version=0.16.0-beta.16
 
 # Mod Properties
 	maven_group = com.example
diff --git a/src/test/resources/projects/maven/build.gradle b/src/test/resources/projects/maven/build.gradle
index 0b19a3addd2b7f9e4dd18d9d3c9f0cd143be5296..31388aa4502daf927fddde416d781130fa6fcf2c 100644
--- a/src/test/resources/projects/maven/build.gradle
+++ b/src/test/resources/projects/maven/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'org.quiltmc.loom'
 	id 'maven-publish'
 }
 
@@ -19,7 +19,7 @@ repositories {
 dependencies {
 	minecraft "com.mojang:minecraft:1.16.5"
 	mappings "net.fabricmc:yarn:1.16.5+build.5:v2"
-	modImplementation "net.fabricmc:fabric-loader:0.11.2"
+	modImplementation "org.quiltmc:quilt-loader:0.16.0-beta.5"
 
 	modImplementation System.getProperty("loom.test.resolve")
 }
diff --git a/src/test/resources/projects/mavenLibrary/build.gradle b/src/test/resources/projects/mavenLibrary/build.gradle
index 0001e2b7dd835ce6e1ef90415bb373563ec8e823..eb3d45b9780c05557162ffefe0cd8c8ec23a37c4 100644
--- a/src/test/resources/projects/mavenLibrary/build.gradle
+++ b/src/test/resources/projects/mavenLibrary/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'org.quiltmc.loom'
 	id 'maven-publish'
 }
 
@@ -12,7 +12,7 @@ println archivesBaseName
 dependencies {
 	minecraft "com.mojang:minecraft:1.16.5"
 	mappings "net.fabricmc:yarn:1.16.5+build.5:v2"
-	modImplementation "net.fabricmc:fabric-loader:0.11.2"
+	modImplementation "org.quiltmc:quilt-loader:0.16.0-beta.5"
 	modCompileOnlyApi fabricApi.module('fabric-biome-api-v1', '0.42.0+1.16')
 	modRuntimeOnly fabricApi.module('fabric-command-api-v1', '0.42.0+1.16')
 }
diff --git a/src/test/resources/projects/minimalBase/build.gradle b/src/test/resources/projects/minimalBase/build.gradle
index 8fbea237ed6254ea326fecc3c6db1a732746a6e0..fa18261b9d7f1a5f7999f27a5be68f59bb002a9c 100644
--- a/src/test/resources/projects/minimalBase/build.gradle
+++ b/src/test/resources/projects/minimalBase/build.gradle
@@ -1,7 +1,7 @@
 // This is used by a range of tests that append to this file before running the gradle tasks.
 // Can be used for tests that require minimal custom setup
 plugins {
-	id 'fabric-loom'
+	id 'org.quiltmc.loom'
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/mixinApAutoRefmap/build.gradle b/src/test/resources/projects/mixinApAutoRefmap/build.gradle
index 610d042d1a47cb4e3a82aad6e147b87248671333..b17bd245b4a9de210ef17cbf9c5b26d698f833b4 100644
--- a/src/test/resources/projects/mixinApAutoRefmap/build.gradle
+++ b/src/test/resources/projects/mixinApAutoRefmap/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'org.quiltmc.loom'
 	id 'com.github.johnrengelman.shadow' version '7.0.0'
 	id 'maven-publish'
 }
@@ -23,7 +23,7 @@ dependencies {
 	// To change the versions see the gradle.properties file
 	minecraft "com.mojang:minecraft:${project.minecraft_version}"
 	mappings "net.fabricmc:yarn:${project.yarn_mappings}:v2"
-	modImplementation "net.fabricmc:fabric-loader:${project.loader_version}"
+	modImplementation "org.quiltmc:quilt-loader:${project.loader_version}"
 
 	// Fabric API. This is technically optional, but you probably want it anyway.
 	modImplementation "net.fabricmc.fabric-api:fabric-api:${project.fabric_version}"
diff --git a/src/test/resources/projects/mixinApAutoRefmap/gradle.properties b/src/test/resources/projects/mixinApAutoRefmap/gradle.properties
index 60cb9a4f2c0b4fca94a187862c8a28afbb2a00cf..0925278e52f63aa39b68cf7aa6b5107541317c20 100644
--- a/src/test/resources/projects/mixinApAutoRefmap/gradle.properties
+++ b/src/test/resources/projects/mixinApAutoRefmap/gradle.properties
@@ -5,7 +5,7 @@ org.gradle.jvmargs=-Xmx1G
 	# check these on https://fabricmc.net/versions.html
 	minecraft_version=1.17
 	yarn_mappings=1.17+build.13
-	loader_version=0.11.6
+	loader_version=0.16.0-beta.16
 
 # Mod Properties
 	mod_version = 1.0.0
diff --git a/src/test/resources/projects/mixinApSimple/build.gradle b/src/test/resources/projects/mixinApSimple/build.gradle
index b21803ea86251da5162273226c5b38e3e4907f3f..c09ae75c545d4fb51d1a54fcd2437d3bfc8e7f9e 100644
--- a/src/test/resources/projects/mixinApSimple/build.gradle
+++ b/src/test/resources/projects/mixinApSimple/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'org.quiltmc.loom'
 	id 'maven-publish'
 }
 
@@ -22,7 +22,7 @@ dependencies {
 	// To change the versions see the gradle.properties file
 	minecraft "com.mojang:minecraft:${project.minecraft_version}"
 	mappings "net.fabricmc:yarn:${project.yarn_mappings}:v2"
-	modImplementation "net.fabricmc:fabric-loader:${project.loader_version}"
+	modImplementation "org.quiltmc:quilt-loader:${project.loader_version}"
 
 	// Fabric API. This is technically optional, but you probably want it anyway.
 	modImplementation "net.fabricmc.fabric-api:fabric-api:${project.fabric_version}"
diff --git a/src/test/resources/projects/mixinApSimple/gradle.properties b/src/test/resources/projects/mixinApSimple/gradle.properties
index 60cb9a4f2c0b4fca94a187862c8a28afbb2a00cf..0925278e52f63aa39b68cf7aa6b5107541317c20 100644
--- a/src/test/resources/projects/mixinApSimple/gradle.properties
+++ b/src/test/resources/projects/mixinApSimple/gradle.properties
@@ -5,7 +5,7 @@ org.gradle.jvmargs=-Xmx1G
 	# check these on https://fabricmc.net/versions.html
 	minecraft_version=1.17
 	yarn_mappings=1.17+build.13
-	loader_version=0.11.6
+	loader_version=0.16.0-beta.16
 
 # Mod Properties
 	mod_version = 1.0.0
diff --git a/src/test/resources/projects/modJavadoc/build.gradle b/src/test/resources/projects/modJavadoc/build.gradle
index 52f1fc39f472e431795ddfe94cd85c6a01eade48..94d0f2d4fa03f8b50dbff269631b3cb6a095b362 100644
--- a/src/test/resources/projects/modJavadoc/build.gradle
+++ b/src/test/resources/projects/modJavadoc/build.gradle
@@ -1,7 +1,7 @@
 // This is used by a range of tests that append to this file before running the gradle tasks.
 // Can be used for tests that require minimal custom setup
 plugins {
-	id 'fabric-loom'
+	id 'org.quiltmc.loom'
 	id 'maven-publish'
 }
 
@@ -12,7 +12,7 @@ group = "com.example"
 dependencies {
 	minecraft "com.mojang:minecraft:1.17.1"
 	mappings "net.fabricmc:yarn:1.17.1+build.59:v2"
-	modImplementation "net.fabricmc:fabric-loader:0.11.6"
+	modImplementation "org.quiltmc:quilt-loader:0.16.0-beta.16"
 
 	modImplementation files("dummy.jar")
 }
\ No newline at end of file
diff --git a/src/test/resources/projects/mojangMappings/build.gradle b/src/test/resources/projects/mojangMappings/build.gradle
index 895bdb752a2aec465a2d430814229a1b2d35b0ee..803040561028fa1860f10de0c1e24011a54a2d69 100644
--- a/src/test/resources/projects/mojangMappings/build.gradle
+++ b/src/test/resources/projects/mojangMappings/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'org.quiltmc.loom'
 	id 'maven-publish'
 }
 
@@ -22,7 +22,7 @@ dependencies {
 	// To change the versions see the gradle.properties file
 	minecraft "com.mojang:minecraft:${project.minecraft_version}"
 	mappings loom.officialMojangMappings()
-	modImplementation "net.fabricmc:fabric-loader:${project.loader_version}"
+	modImplementation "org.quiltmc:quilt-loader:${project.loader_version}"
 
 	// Fabric API. This is technically optional, but you probably want it anyway.
 	modImplementation "net.fabricmc.fabric-api:fabric-api:${project.fabric_version}"
diff --git a/src/test/resources/projects/mojangMappings/gradle.properties b/src/test/resources/projects/mojangMappings/gradle.properties
index 2e85dfd8d7d617d3e8d5c71e0eeab4c93891edbf..970667432831ef16d900cbb1822786e61c804260 100644
--- a/src/test/resources/projects/mojangMappings/gradle.properties
+++ b/src/test/resources/projects/mojangMappings/gradle.properties
@@ -5,7 +5,7 @@ org.gradle.jvmargs=-Xmx1G
 	# check these on https://fabricmc.net/use
 	minecraft_version=1.16.5
 	yarn_mappings=1.16.5+build.5
-	loader_version=0.11.2
+	loader_version=0.16.0-beta.16
 
 # Mod Properties
 	mod_version = 1.0.0
diff --git a/src/test/resources/projects/multiproject/build.gradle b/src/test/resources/projects/multiproject/build.gradle
index 62858e2f46a1c7d4e7cbc5fe7337bb90961c8490..f9fc1afa54be944b364d021cd4dc007bfac4f84a 100644
--- a/src/test/resources/projects/multiproject/build.gradle
+++ b/src/test/resources/projects/multiproject/build.gradle
@@ -1,11 +1,11 @@
 plugins {
 	id "java"
 	id "java-library"
-	id 'fabric-loom' apply false
+	id 'org.quiltmc.loom' apply false
 }
 
 allprojects {
-	apply plugin: "fabric-loom"
+	apply plugin: "org.quiltmc.loom"
 
 	version = "1.0.0"
 
@@ -25,7 +25,7 @@ allprojects {
 		// To change the versions see the gradle.properties file
 		minecraft "com.mojang:minecraft:1.16.5"
 		mappings "net.fabricmc:yarn:1.16.5+build.5:v2"
-		modImplementation "net.fabricmc:fabric-loader:0.11.2"
+		modImplementation "org.quiltmc:quilt-loader:0.16.0-beta.5"
 
 		modImplementation "net.fabricmc.fabric-api:fabric-api:0.31.0+1.16"
 
diff --git a/src/test/resources/projects/multiproject/gradle.properties b/src/test/resources/projects/multiproject/gradle.properties
index 2e85dfd8d7d617d3e8d5c71e0eeab4c93891edbf..970667432831ef16d900cbb1822786e61c804260 100644
--- a/src/test/resources/projects/multiproject/gradle.properties
+++ b/src/test/resources/projects/multiproject/gradle.properties
@@ -5,7 +5,7 @@ org.gradle.jvmargs=-Xmx1G
 	# check these on https://fabricmc.net/use
 	minecraft_version=1.16.5
 	yarn_mappings=1.16.5+build.5
-	loader_version=0.11.2
+	loader_version=0.16.0-beta.16
 
 # Mod Properties
 	mod_version = 1.0.0
diff --git a/src/test/resources/projects/parchment/build.gradle b/src/test/resources/projects/parchment/build.gradle
index 5fee81c5d1e07de0dfcf615795ec818a8e4705e2..76ec0e30196bfae2f5b92b2e8b722256d980da0e 100644
--- a/src/test/resources/projects/parchment/build.gradle
+++ b/src/test/resources/projects/parchment/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'org.quiltmc.loom'
 }
 
 repositories {
@@ -18,5 +18,5 @@ dependencies {
 		}
 	}
 
-	modImplementation libs.fabricLoader
+	modImplementation libs.loader
 }
diff --git a/src/test/resources/projects/parchment/gradle/libs.versions.toml b/src/test/resources/projects/parchment/gradle/libs.versions.toml
index 40c22f786bae76ab2365e2a337d8a25486a5b9e3..cd3bac3e45d8aceb1acba8ace079bfce926b4f64 100644
--- a/src/test/resources/projects/parchment/gradle/libs.versions.toml
+++ b/src/test/resources/projects/parchment/gradle/libs.versions.toml
@@ -1,9 +1,9 @@
 [versions]
 minecraft = "1.16.5"
-fabricLoader = "0.11.3"
+loader = "0.16.0-beta.5"
 parchment = "2021.10.10"
 
 [libraries]
 minecraft = { module = "com.mojang:minecraft", version.ref = "minecraft" }
-fabricLoader = { module = "net.fabricmc:fabric-loader", version.ref = "fabricLoader" }
+loader = { module = "org.quiltmc:quilt-loader", version.ref = "loader" }
 parchment = { module = "org.parchmentmc.data:parchment-1.16.5", version.ref = "parchment" }
\ No newline at end of file
diff --git a/src/test/resources/projects/reproducible/build.gradle b/src/test/resources/projects/reproducible/build.gradle
index 4bc4926136fa448fd4268d7392217bf7f160fb9a..038f6fd61e6952a6a124d61c6ccb410da4dd9079 100644
--- a/src/test/resources/projects/reproducible/build.gradle
+++ b/src/test/resources/projects/reproducible/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'org.quiltmc.loom'
 	id 'maven-publish'
 }
 
@@ -19,7 +19,7 @@ loom {
 dependencies {
 	minecraft "com.mojang:minecraft:${project.minecraft_version}"
 	mappings "net.fabricmc:yarn:${project.yarn_mappings}:v2"
-	modImplementation "net.fabricmc:fabric-loader:${project.loader_version}"
+	modImplementation "org.quiltmc:quilt-loader:${project.loader_version}"
 }
 
 processResources {
diff --git a/src/test/resources/projects/reproducible/gradle.properties b/src/test/resources/projects/reproducible/gradle.properties
index 3dc7831a9262fea0e3779896665bdc7a8c979f16..ce2badeb0f1e4860637304a0ba79f59d5fe0473e 100644
--- a/src/test/resources/projects/reproducible/gradle.properties
+++ b/src/test/resources/projects/reproducible/gradle.properties
@@ -5,7 +5,7 @@ org.gradle.jvmargs=-Xmx1G
 	# check these on https://fabricmc.net/use
 	minecraft_version=1.16.5
 	yarn_mappings=1.16.5+build.5
-	loader_version=0.11.2
+	loader_version=0.16.0-beta.16
 
 # Mod Properties
 	mod_version = 1.0.0
diff --git a/src/test/resources/projects/runconfigs/build.gradle b/src/test/resources/projects/runconfigs/build.gradle
index a8e1f6bb073f57d6196eebe782836ed3f436b19b..23048574367f5a5bff9cb04b16862ba4a33a9a8b 100644
--- a/src/test/resources/projects/runconfigs/build.gradle
+++ b/src/test/resources/projects/runconfigs/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'org.quiltmc.loom'
 }
 
 loom {
@@ -33,5 +33,5 @@ version = "1.0.0"
 dependencies {
 	minecraft "com.mojang:minecraft:1.18.1"
 	mappings "net.fabricmc:yarn:1.18.1+build.12:v2"
-	modImplementation "net.fabricmc:fabric-loader:0.12.12"
+	modImplementation "org.quiltmc:quilt-loader:0.16.0-beta.5"
 }
\ No newline at end of file
diff --git a/src/test/resources/projects/signed/build.gradle b/src/test/resources/projects/signed/build.gradle
index 45a2a278a761e5f9cc0f56bb7972719196660cfe..f9063a5286bb91e0faa8ff5df85453cc764b6f95 100644
--- a/src/test/resources/projects/signed/build.gradle
+++ b/src/test/resources/projects/signed/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'org.quiltmc.loom'
 	id 'maven-publish'
 	id 'signing'
 }
@@ -11,7 +11,7 @@ group = "com.example"
 dependencies {
 	minecraft "com.mojang:minecraft:1.16.5"
 	mappings "net.fabricmc:yarn:1.16.5+build.5:v2"
-	modImplementation "net.fabricmc:fabric-loader:0.11.2"
+	modImplementation "org.quiltmc:quilt-loader:0.16.0-beta.5"
 }
 
 processResources {
diff --git a/src/test/resources/projects/simple/build.gradle b/src/test/resources/projects/simple/build.gradle
index 5ce232ea8a8d7975252e63da3bb7a471e22c39b3..6e25e95631e5961cd65217cc54171c015676dd3c 100644
--- a/src/test/resources/projects/simple/build.gradle
+++ b/src/test/resources/projects/simple/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'org.quiltmc.loom'
 	id 'maven-publish'
 }
 
@@ -22,7 +22,7 @@ dependencies {
 	// To change the versions see the gradle.properties file
 	minecraft "com.mojang:minecraft:${project.minecraft_version}"
 	mappings "net.fabricmc:yarn:${project.yarn_mappings}:v2"
-	modImplementation "net.fabricmc:fabric-loader:${project.loader_version}"
+	modImplementation "org.quiltmc:quilt-loader:${project.loader_version}"
 
 	// Fabric API. This is technically optional, but you probably want it anyway.
 	modImplementation "net.fabricmc.fabric-api:fabric-api:${project.fabric_version}"
diff --git a/src/test/resources/projects/simple/gradle.properties b/src/test/resources/projects/simple/gradle.properties
index 845c356072dbb73ddb31b97826e8cb4fbaf138e0..c01c88366c6038008ee1d2f668b63bc2de2e138a 100644
--- a/src/test/resources/projects/simple/gradle.properties
+++ b/src/test/resources/projects/simple/gradle.properties
@@ -5,7 +5,7 @@ org.gradle.jvmargs=-Xmx1G
 	# check these on https://fabricmc.net/use
 	minecraft_version=1.16.5
 	yarn_mappings=1.16.5+build.5
-	loader_version=0.11.2
+	loader_version=0.16.0-beta.16
 
 # Mod Properties
 	maven_group = com.example
diff --git a/src/test/resources/projects/splitSources/build.gradle b/src/test/resources/projects/splitSources/build.gradle
index c4d5174f802499c7e6a6bddd3b1c7f45ea542fc7..450500ff6e7777ed4264b5150eb86bfd4b74ca2f 100644
--- a/src/test/resources/projects/splitSources/build.gradle
+++ b/src/test/resources/projects/splitSources/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom' version '0.12.local'
+	id 'org.quiltmc.loom' version '0.12.local'
 	id 'maven-publish'
 }
 
@@ -26,7 +26,7 @@ loom {
 dependencies {
 	minecraft "com.mojang:minecraft:1.18.2"
 	mappings "net.fabricmc:yarn:1.18.2+build.1:v2"
-	modImplementation "net.fabricmc:fabric-loader:0.13.3"
+	modImplementation "org.quiltmc:quilt-loader:0.16.0-beta.5"
 
 	modImplementation "net.fabricmc.fabric-api:fabric-api:0.47.8+1.18.2"
 
diff --git a/src/test/resources/projects/transitiveAccesswidener/build.gradle b/src/test/resources/projects/transitiveAccesswidener/build.gradle
index 52f1fc39f472e431795ddfe94cd85c6a01eade48..301cfe010e0dcff7226d980f2c2c69e5a589e702 100644
--- a/src/test/resources/projects/transitiveAccesswidener/build.gradle
+++ b/src/test/resources/projects/transitiveAccesswidener/build.gradle
@@ -1,7 +1,7 @@
 // This is used by a range of tests that append to this file before running the gradle tasks.
 // Can be used for tests that require minimal custom setup
 plugins {
-	id 'fabric-loom'
+	id 'org.quiltmc.loom'
 	id 'maven-publish'
 }
 
@@ -12,7 +12,7 @@ group = "com.example"
 dependencies {
 	minecraft "com.mojang:minecraft:1.17.1"
 	mappings "net.fabricmc:yarn:1.17.1+build.59:v2"
-	modImplementation "net.fabricmc:fabric-loader:0.11.6"
+	modImplementation "org.quiltmc:quilt-loader:0.16.0-beta.5"
 
 	modImplementation files("dummy.jar")
 }
\ No newline at end of file
diff --git a/src/test/resources/projects/unpick/build.gradle b/src/test/resources/projects/unpick/build.gradle
index 4b79a1b01f6d59fea5174e912c213a1ee597327f..3d4edf7fe25337ea5db31367ec65cf207ac98f9b 100644
--- a/src/test/resources/projects/unpick/build.gradle
+++ b/src/test/resources/projects/unpick/build.gradle
@@ -1,9 +1,9 @@
 plugins {
-	id 'fabric-loom'
+	id 'org.quiltmc.loom'
 }
 
 dependencies {
 	minecraft "com.mojang:minecraft:21w13a"
 	mappings "net.fabricmc:yarn:21w13a+build.30:v2"
-	modImplementation "net.fabricmc:fabric-loader:0.11.3"
+	modImplementation "org.quiltmc:quilt-loader:0.16.0-beta.5"
 }
